erts/mac/man/man3/CosNotifyFilter_Filter.3
.TH CosNotifyFilter_Filter 3 "cosNotification 1.2" "Ericsson AB" "Erlang Module Definition"
.SH NAME
CosNotifyFilter_Filter \- This module implements the OMG CosNotifyFilter::Filter interface.
.SH DESCRIPTION
.LP
To get access to the record definitions for the structures use: 
.br
\fI-include_lib("cosNotification/include/*\&.hrl")\&.\fR\&
.SH EXPORTS
.LP
.B
_get_constraint_grammar(Filter) -> Grammar
.br
.RS
.LP
Types:

.RS 3
Filter = #objref
.br
Grammar = string()
.br
.RE
.RE
.RS
.LP
This operation returns which type of Grammar the Filter uses\&. Currently, only \fI"EXTENDED_TCL"\fR\& is supported\&.
.RE
.LP
.B
add_constraints(Filter, ConstraintExpSeq) -> Reply
.br
.RS
.LP
Types:

.RS 3
Filter = #objref
.br
ConstraintExpSeq = [Constraint]
.br
ConstraintExp = #\&'CosNotifyFilter_ConstraintExp\&'{event_types, constraint_expr}
.br
event_types = #\&'CosNotification_EventTypeSeq\&'{}
.br
constraint_expr = string()
.br
Reply = ConstraintInfoSeq | {\&'EXCEPTION\&', #\&'CosNotifyFilter_InvalidConstraint\&'{constr}}
.br
constr = ConstraintExp
.br
ConstraintInfoSeq = [ConstraintInfo]
.br
ConstraintInfo = #\&'CosNotifyFilter_ConstraintInfo\&'{constraint_expression, constraint_id}
.br
constraint_expression = ConstraintExp
.br
constraint_id = long()
.br
.RE
.RE
.RS
.LP
Initially, Filters do not contain any constraints, hence, all events will be forwarded\&. The \fIadd_constraints/2\fR\& operation allow us to add constraints to the target object\&.
.RE
.LP
.B
modify_constraints(Filter, ConstraintIDSeq, ConstraintInfoSeq) -> Reply
.br
.RS
.LP
Types:

.RS 3
Filter = #objref
.br
ConstraintIDSeq = [ConstraintID]
.br
ConstraintID = long()
.br
ConstraintInfoSeq = [ConstraintInfo]
.br
ConstraintInfo = #\&'CosNotifyFilter_ConstraintInfo\&'{constraint_expression, constraint_id}
.br
constraint_expression = ConstraintExp
.br
constraint_id = long()
.br
Reply = ok | {\&'EXCEPTION\&', #\&'CosNotifyFilter_InvalidConstraint\&'{constr}} | {\&'EXCEPTION\&', #\&'CosNotifyFilter_ConstraintNotFound\&'{id}}
.br
constr = ConstraintExp
.br
id = long()
.br
ConstraintExp = #\&'CosNotifyFilter_ConstraintExp\&'{event_types, constraint_expr}
.br
event_types = #\&'CosNotification_EventTypeSeq\&'{}
.br
constraint_expr = string()
.br
.RE
.RE
.RS
.LP
This operation is invoked by a client in order to modify the constraints associated with the target object\&. The constraints related to the Id\&'s in the parameter sequence \fIConstraintIDSeq\fR\& will, if all values are valid, be deleted\&. The \fIConstraintInfoSeq\fR\& parameter contains of Id-Expression pairs and a constraint matching one of the unique Id\&'s will, if all input values are correct, be updated\&. If the parameters contain incorrect data en exception will be raised\&.
.RE
.LP
.B
get_constraints(Filter, ConstraintIDSeq) -> Reply
.br
.RS
.LP
Types:

.RS 3
Filter = #objref
.br
ConstraintIDSeq = [ConstraintID]
.br
ConstraintID = long()
.br
Reply = ConstraintInfoSeq | {\&'EXCEPTION\&', #\&'CosNotifyFilter_ConstraintNotFound\&'{id}}
.br
ConstraintInfoSeq = [ConstraintInfo]
.br
ConstraintInfo = #\&'CosNotifyFilter_ConstraintInfo\&'{constraint_expression, constraint_id}
.br
constraint_expression = ConstraintExp
.br
constraint_id = id = long()
.br
.RE
.RE
.RS
.LP
This operation return a sequence of ConstraintInfo\&'s, related to the given ConstraintID\&'s, associated with the target object\&.
.RE
.LP
.B
get_all_constraints(Filter) -> ConstraintInfoSeq
.br
.RS
.LP
Types:

.RS 3
Filter = #objref
.br
ConstraintInfoSeq = [ConstraintInfo]
.br
ConstraintInfo = #\&'CosNotifyFilter_ConstraintInfo\&'{constraint_expression, constraint_id}
.br
constraint_expression = ConstraintExp
.br
constraint_id = long()
.br
.RE
.RE
.RS
.LP
All constraints, and their unique Id, associated with the target object will be returned by this operation\&.
.RE
.LP
.B
remove_all_constraints(Filter) -> ok
.br
.RS
.LP
Types:

.RS 3
Filter = #objref
.br
.RE
.RE
.RS
.LP
All constraints associated with the target object are removed by this operation and, since the the target object no longer contain any constraints, true will always be the result of any match operation\&.
.RE
.LP
.B
destroy(Filter) -> ok
.br
.RS
.LP
Types:

.RS 3
Filter = #objref
.br
.RE
.RE
.RS
.LP
This operation terminates the target object\&.
.RE
.LP
.B
match(Filter, Event) -> Reply
.br
.RS
.LP
Types:

.RS 3
Filter = #objref
.br
Event = #any
.br
Reply = boolean() | {\&'EXCEPTION\&', #\&'CosNotifyFilter_UnsupportedFilterableData\&'{}}
.br
.RE
.RE
.RS
.LP
This operation accepts an \fI#any{}\fR\& event and returns \fItrue\fR\& if it satisfies at least one constraint\&. If the event contains data of the wrong type, e\&.g\&., should be a string() but in fact i a short(), an exception is raised\&.
.RE
.LP
.B
match_structured(Filter, Event) -> Reply
.br
.RS
.LP
Types:

.RS 3
Filter = #objref
.br
Event = #\&'CosNotification_StructuredEvent\&'{}
.br
Reply = boolean() | {\&'EXCEPTION\&', #\&'CosNotifyFilter_UnsupportedFilterableData\&'{}}
.br
.RE
.RE
.RS
.LP
This operation is similar to the \fImatch\fR\& operation but accepts structured events instead\&.
.RE
.LP
.B
attach_callback(Filter, NotifySubscribe) -> CallbackID
.br
.RS
.LP
Types:

.RS 3
Filter = #objref
.br
NotifySubscribe = #objref
.br
CallbackID = long()
.br
.RE
.RE
.RS
.LP
This operation connects a NotifySubscribe object, which should be informed when the target object\&'s constraints are updated\&. A unique Id is returned which must be stored if we ever want to detach the callback object in the future\&.
.RE
.LP
.B
detach_callback(Filter, CallbackID) -> Reply
.br
.RS
.LP
Types:

.RS 3
Filter = #objref
.br
CallbackID = long()
.br
Reply = ok | {\&'EXCEPTION\&', #\&'CosNotifyFilter_CallbackNotFound\&'{}}
.br
.RE
.RE
.RS
.LP
If the target object has an associated callback that matches the supplied Id it will be removed and longer informed of any updates\&. If no object with a matching Id is found an exception is raised\&.
.RE
.LP
.B
get_callbacks(Filter) -> CallbackIDSeq
.br
.RS
.LP
Types:

.RS 3
Filter = #objref
.br
CallbackIDSeq = [CallbackID]
.br
CallbackID = long()
.br
.RE
.RE
.RS
.LP
This operation returns a sequence of all connected NotifySubscribe object Id\&'s\&. If no callbacks are associated with the target object the list will be empty\&.
.RE

erts/mac/lib/cosNotification-1.2/src/CosNotifyFilter_Filter.erl
%%  coding: latin-1
%%------------------------------------------------------------
%%
%% Implementation stub file
%% 
%% Target: CosNotifyFilter_Filter
%% Source: /private/tmp/erlang20150820-34163-1k9di6g/otp-OTP-18.0.3/lib/cosNotification/src/CosNotifyFilter.idl
%% IC vsn: 4.4
%% 
%% This file is automatically generated. DO NOT EDIT IT.
%%
%%------------------------------------------------------------

-module('CosNotifyFilter_Filter').
-ic_compiled("4_4").


%% Interface functions
-export(['_get_constraint_grammar'/1, '_get_constraint_grammar'/2, add_constraints/2]).
-export([add_constraints/3, modify_constraints/3, modify_constraints/4]).
-export([get_constraints/2, get_constraints/3, get_all_constraints/1]).
-export([get_all_constraints/2, remove_all_constraints/1, remove_all_constraints/2]).
-export([destroy/1, destroy/2, match/2]).
-export([match/3, match_structured/2, match_structured/3]).
-export([match_typed/2, match_typed/3, attach_callback/2]).
-export([attach_callback/3, detach_callback/2, detach_callback/3]).
-export([get_callbacks/1, get_callbacks/2]).

%% Type identification function
-export([typeID/0]).

%% Used to start server
-export([oe_create/0, oe_create_link/0, oe_create/1]).
-export([oe_create_link/1, oe_create/2, oe_create_link/2]).

%% TypeCode Functions and inheritance
-export([oe_tc/1, oe_is_a/1, oe_get_interface/0]).

%% gen server export stuff
-behaviour(gen_server).
-export([init/1, terminate/2, handle_call/3]).
-export([handle_cast/2, handle_info/2, code_change/3]).

-include_lib("orber/include/corba.hrl").


%%------------------------------------------------------------
%%
%% Object interface functions.
%%
%%------------------------------------------------------------



%%%% Operation: '_get_constraint_grammar'
%% 
%%   Returns: RetVal
%%
'_get_constraint_grammar'(OE_THIS) ->
    corba:call(OE_THIS, '_get_constraint_grammar', [], ?MODULE).

'_get_constraint_grammar'(OE_THIS, OE_Options) ->
    corba:call(OE_THIS, '_get_constraint_grammar', [], ?MODULE, OE_Options).

%%%% Operation: add_constraints
%% 
%%   Returns: RetVal
%%   Raises:  CosNotifyFilter::InvalidConstraint
%%
add_constraints(OE_THIS, Constraint_list) ->
    corba:call(OE_THIS, add_constraints, [Constraint_list], ?MODULE).

add_constraints(OE_THIS, OE_Options, Constraint_list) ->
    corba:call(OE_THIS, add_constraints, [Constraint_list], ?MODULE, OE_Options).

%%%% Operation: modify_constraints
%% 
%%   Returns: RetVal
%%   Raises:  CosNotifyFilter::InvalidConstraint, CosNotifyFilter::ConstraintNotFound
%%
modify_constraints(OE_THIS, Del_list, Modify_list) ->
    corba:call(OE_THIS, modify_constraints, [Del_list, Modify_list], ?MODULE).

modify_constraints(OE_THIS, OE_Options, Del_list, Modify_list) ->
    corba:call(OE_THIS, modify_constraints, [Del_list, Modify_list], ?MODULE, OE_Options).

%%%% Operation: get_constraints
%% 
%%   Returns: RetVal
%%   Raises:  CosNotifyFilter::ConstraintNotFound
%%
get_constraints(OE_THIS, Id_list) ->
    corba:call(OE_THIS, get_constraints, [Id_list], ?MODULE).

get_constraints(OE_THIS, OE_Options, Id_list) ->
    corba:call(OE_THIS, get_constraints, [Id_list], ?MODULE, OE_Options).

%%%% Operation: get_all_constraints
%% 
%%   Returns: RetVal
%%
get_all_constraints(OE_THIS) ->
    corba:call(OE_THIS, get_all_constraints, [], ?MODULE).

get_all_constraints(OE_THIS, OE_Options) ->
    corba:call(OE_THIS, get_all_constraints, [], ?MODULE, OE_Options).

%%%% Operation: remove_all_constraints
%% 
%%   Returns: RetVal
%%
remove_all_constraints(OE_THIS) ->
    corba:call(OE_THIS, remove_all_constraints, [], ?MODULE).

remove_all_constraints(OE_THIS, OE_Options) ->
    corba:call(OE_THIS, remove_all_constraints, [], ?MODULE, OE_Options).

%%%% Operation: destroy
%% 
%%   Returns: RetVal
%%
destroy(OE_THIS) ->
    corba:call(OE_THIS, destroy, [], ?MODULE).

destroy(OE_THIS, OE_Options) ->
    corba:call(OE_THIS, destroy, [], ?MODULE, OE_Options).

%%%% Operation: match
%% 
%%   Returns: RetVal
%%   Raises:  CosNotifyFilter::UnsupportedFilterableData
%%
match(OE_THIS, Filterable_data) ->
    corba:call(OE_THIS, match, [Filterable_data], ?MODULE).

match(OE_THIS, OE_Options, Filterable_data) ->
    corba:call(OE_THIS, match, [Filterable_data], ?MODULE, OE_Options).

%%%% Operation: match_structured
%% 
%%   Returns: RetVal
%%   Raises:  CosNotifyFilter::UnsupportedFilterableData
%%
match_structured(OE_THIS, Filterable_data) ->
    corba:call(OE_THIS, match_structured, [Filterable_data], ?MODULE).

match_structured(OE_THIS, OE_Options, Filterable_data) ->
    corba:call(OE_THIS, match_structured, [Filterable_data], ?MODULE, OE_Options).

%%%% Operation: match_typed
%% 
%%   Returns: RetVal
%%   Raises:  CosNotifyFilter::UnsupportedFilterableData
%%
match_typed(OE_THIS, Filterable_data) ->
    corba:call(OE_THIS, match_typed, [Filterable_data], ?MODULE).

match_typed(OE_THIS, OE_Options, Filterable_data) ->
    corba:call(OE_THIS, match_typed, [Filterable_data], ?MODULE, OE_Options).

%%%% Operation: attach_callback
%% 
%%   Returns: RetVal
%%
attach_callback(OE_THIS, Callback) ->
    corba:call(OE_THIS, attach_callback, [Callback], ?MODULE).

attach_callback(OE_THIS, OE_Options, Callback) ->
    corba:call(OE_THIS, attach_callback, [Callback], ?MODULE, OE_Options).

%%%% Operation: detach_callback
%% 
%%   Returns: RetVal
%%   Raises:  CosNotifyFilter::CallbackNotFound
%%
detach_callback(OE_THIS, Callback) ->
    corba:call(OE_THIS, detach_callback, [Callback], ?MODULE).

detach_callback(OE_THIS, OE_Options, Callback) ->
    corba:call(OE_THIS, detach_callback, [Callback], ?MODULE, OE_Options).

%%%% Operation: get_callbacks
%% 
%%   Returns: RetVal
%%
get_callbacks(OE_THIS) ->
    corba:call(OE_THIS, get_callbacks, [], ?MODULE).

get_callbacks(OE_THIS, OE_Options) ->
    corba:call(OE_THIS, get_callbacks, [], ?MODULE, OE_Options).

%%------------------------------------------------------------
%%
%% Inherited Interfaces
%%
%%------------------------------------------------------------
oe_is_a("IDL:omg.org/CosNotifyFilter/Filter:1.0") -> true;
oe_is_a(_) -> false.

%%------------------------------------------------------------
%%
%% Interface TypeCode
%%
%%------------------------------------------------------------
oe_tc('_get_constraint_grammar') -> 
	{{tk_string,0},[],[]};
oe_tc(add_constraints) -> 
	{{tk_sequence,
             {tk_struct,"IDL:omg.org/CosNotifyFilter/ConstraintInfo:1.0",
                 "ConstraintInfo",
                 [{"constraint_expression",
                   {tk_struct,
                       "IDL:omg.org/CosNotifyFilter/ConstraintExp:1.0",
                       "ConstraintExp",
                       [{"event_types",
                         {tk_sequence,
                             {tk_struct,
                                 "IDL:omg.org/CosNotification/EventType:1.0",
                                 "EventType",
                                 [{"domain_name",{tk_string,0}},
                                  {"type_name",{tk_string,0}}]},
                             0}},
                        {"constraint_expr",{tk_string,0}}]}},
                  {"constraint_id",tk_long}]},
             0},
         [{tk_sequence,
              {tk_struct,"IDL:omg.org/CosNotifyFilter/ConstraintExp:1.0",
                  "ConstraintExp",
                  [{"event_types",
                    {tk_sequence,
                        {tk_struct,
                            "IDL:omg.org/CosNotification/EventType:1.0",
                            "EventType",
                            [{"domain_name",{tk_string,0}},
                             {"type_name",{tk_string,0}}]},
                        0}},
                   {"constraint_expr",{tk_string,0}}]},
              0}],
         []};
oe_tc(modify_constraints) -> 
	{tk_void,
            [{tk_sequence,tk_long,0},
             {tk_sequence,
                 {tk_struct,"IDL:omg.org/CosNotifyFilter/ConstraintInfo:1.0",
                     "ConstraintInfo",
                     [{"constraint_expression",
                       {tk_struct,
                           "IDL:omg.org/CosNotifyFilter/ConstraintExp:1.0",
                           "ConstraintExp",
                           [{"event_types",
                             {tk_sequence,
                                 {tk_struct,
                                     "IDL:omg.org/CosNotification/EventType:1.0",
                                     "EventType",
                                     [{"domain_name",{tk_string,0}},
                                      {"type_name",{tk_string,0}}]},
                                 0}},
                            {"constraint_expr",{tk_string,0}}]}},
                      {"constraint_id",tk_long}]},
                 0}],
            []};
oe_tc(get_constraints) -> 
	{{tk_sequence,
             {tk_struct,"IDL:omg.org/CosNotifyFilter/ConstraintInfo:1.0",
                 "ConstraintInfo",
                 [{"constraint_expression",
                   {tk_struct,
                       "IDL:omg.org/CosNotifyFilter/ConstraintExp:1.0",
                       "ConstraintExp",
                       [{"event_types",
                         {tk_sequence,
                             {tk_struct,
                                 "IDL:omg.org/CosNotification/EventType:1.0",
                                 "EventType",
                                 [{"domain_name",{tk_string,0}},
                                  {"type_name",{tk_string,0}}]},
                             0}},
                        {"constraint_expr",{tk_string,0}}]}},
                  {"constraint_id",tk_long}]},
             0},
         [{tk_sequence,tk_long,0}],
         []};
oe_tc(get_all_constraints) -> 
	{{tk_sequence,
             {tk_struct,"IDL:omg.org/CosNotifyFilter/ConstraintInfo:1.0",
                 "ConstraintInfo",
                 [{"constraint_expression",
                   {tk_struct,
                       "IDL:omg.org/CosNotifyFilter/ConstraintExp:1.0",
                       "ConstraintExp",
                       [{"event_types",
                         {tk_sequence,
                             {tk_struct,
                                 "IDL:omg.org/CosNotification/EventType:1.0",
                                 "EventType",
                                 [{"domain_name",{tk_string,0}},
                                  {"type_name",{tk_string,0}}]},
                             0}},
                        {"constraint_expr",{tk_string,0}}]}},
                  {"constraint_id",tk_long}]},
             0},
         [],[]};
oe_tc(remove_all_constraints) -> 
	{tk_void,[],[]};
oe_tc(destroy) -> 
	{tk_void,[],[]};
oe_tc(match) -> 
	{tk_boolean,[tk_any],[]};
oe_tc(match_structured) -> 
	{tk_boolean,
            [{tk_struct,"IDL:omg.org/CosNotification/StructuredEvent:1.0",
                 "StructuredEvent",
                 [{"header",
                   {tk_struct,"IDL:omg.org/CosNotification/EventHeader:1.0",
                       "EventHeader",
                       [{"fixed_header",
                         {tk_struct,
                             "IDL:omg.org/CosNotification/FixedEventHeader:1.0",
                             "FixedEventHeader",
                             [{"event_type",
                               {tk_struct,
                                   "IDL:omg.org/CosNotification/EventType:1.0",
                                   "EventType",
                                   [{"domain_name",{tk_string,0}},
                                    {"type_name",{tk_string,0}}]}},
                              {"event_name",{tk_string,0}}]}},
                        {"variable_header",
                         {tk_sequence,
                             {tk_struct,
                                 "IDL:omg.org/CosNotification/Property:1.0",
                                 "Property",
                                 [{"name",{tk_string,0}},{"value",tk_any}]},
                             0}}]}},
                  {"filterable_data",
                   {tk_sequence,
                       {tk_struct,"IDL:omg.org/CosNotification/Property:1.0",
                           "Property",
                           [{"name",{tk_string,0}},{"value",tk_any}]},
                       0}},
                  {"remainder_of_body",tk_any}]}],
            []};
oe_tc(match_typed) -> 
	{tk_boolean,
            [{tk_sequence,
                 {tk_struct,"IDL:omg.org/CosNotification/Property:1.0",
                     "Property",
                     [{"name",{tk_string,0}},{"value",tk_any}]},
                 0}],
            []};
oe_tc(attach_callback) -> 
	{tk_long,[{tk_objref,"IDL:omg.org/CosNotifyComm/NotifySubscribe:1.0",
                             "NotifySubscribe"}],
                 []};
oe_tc(detach_callback) -> 
	{tk_void,[tk_long],[]};
oe_tc(get_callbacks) -> 
	{{tk_sequence,tk_long,0},[],[]};
oe_tc(_) -> undefined.

oe_get_interface() -> 
	[{"get_callbacks", oe_tc(get_callbacks)},
	{"detach_callback", oe_tc(detach_callback)},
	{"attach_callback", oe_tc(attach_callback)},
	{"match_typed", oe_tc(match_typed)},
	{"match_structured", oe_tc(match_structured)},
	{"match", oe_tc(match)},
	{"destroy", oe_tc(destroy)},
	{"remove_all_constraints", oe_tc(remove_all_constraints)},
	{"get_all_constraints", oe_tc(get_all_constraints)},
	{"get_constraints", oe_tc(get_constraints)},
	{"modify_constraints", oe_tc(modify_constraints)},
	{"add_constraints", oe_tc(add_constraints)},
	{"_get_constraint_grammar", oe_tc('_get_constraint_grammar')}].




%%------------------------------------------------------------
%%
%% Object server implementation.
%%
%%------------------------------------------------------------


%%------------------------------------------------------------
%%
%% Function for fetching the interface type ID.
%%
%%------------------------------------------------------------

typeID() ->
    "IDL:omg.org/CosNotifyFilter/Filter:1.0".


%%------------------------------------------------------------
%%
%% Object creation functions.
%%
%%------------------------------------------------------------

oe_create() ->
    corba:create(?MODULE, "IDL:omg.org/CosNotifyFilter/Filter:1.0").

oe_create_link() ->
    corba:create_link(?MODULE, "IDL:omg.org/CosNotifyFilter/Filter:1.0").

oe_create(Env) ->
    corba:create(?MODULE, "IDL:omg.org/CosNotifyFilter/Filter:1.0", Env).

oe_create_link(Env) ->
    corba:create_link(?MODULE, "IDL:omg.org/CosNotifyFilter/Filter:1.0", Env).

oe_create(Env, RegName) ->
    corba:create(?MODULE, "IDL:omg.org/CosNotifyFilter/Filter:1.0", Env, RegName).

oe_create_link(Env, RegName) ->
    corba:create_link(?MODULE, "IDL:omg.org/CosNotifyFilter/Filter:1.0", Env, RegName).

%%------------------------------------------------------------
%%
%% Init & terminate functions.
%%
%%------------------------------------------------------------

init(Env) ->
%% Call to implementation init
    corba:handle_init('CosNotifyFilter_Filter_impl', Env).

terminate(Reason, State) ->
    corba:handle_terminate('CosNotifyFilter_Filter_impl', Reason, State).


%%%% Operation: '_get_constraint_grammar'
%% 
%%   Returns: RetVal
%%
handle_call({OE_THIS, OE_Context, '_get_constraint_grammar', []}, _, OE_State) ->
  corba:handle_call('CosNotifyFilter_Filter_impl', '_get_constraint_grammar', [], OE_State, OE_Context, OE_THIS, false);

%%%% Operation: add_constraints
%% 
%%   Returns: RetVal
%%   Raises:  CosNotifyFilter::InvalidConstraint
%%
handle_call({OE_THIS, OE_Context, add_constraints, [Constraint_list]}, _, OE_State) ->
  corba:handle_call('CosNotifyFilter_Filter_impl', add_constraints, [Constraint_list], OE_State, OE_Context, OE_THIS, false);

%%%% Operation: modify_constraints
%% 
%%   Returns: RetVal
%%   Raises:  CosNotifyFilter::InvalidConstraint, CosNotifyFilter::ConstraintNotFound
%%
handle_call({OE_THIS, OE_Context, modify_constraints, [Del_list, Modify_list]}, _, OE_State) ->
  corba:handle_call('CosNotifyFilter_Filter_impl', modify_constraints, [Del_list, Modify_list], OE_State, OE_Context, OE_THIS, false);

%%%% Operation: get_constraints
%% 
%%   Returns: RetVal
%%   Raises:  CosNotifyFilter::ConstraintNotFound
%%
handle_call({OE_THIS, OE_Context, get_constraints, [Id_list]}, _, OE_State) ->
  corba:handle_call('CosNotifyFilter_Filter_impl', get_constraints, [Id_list], OE_State, OE_Context, OE_THIS, false);

%%%% Operation: get_all_constraints
%% 
%%   Returns: RetVal
%%
handle_call({OE_THIS, OE_Context, get_all_constraints, []}, _, OE_State) ->
  corba:handle_call('CosNotifyFilter_Filter_impl', get_all_constraints, [], OE_State, OE_Context, OE_THIS, false);

%%%% Operation: remove_all_constraints
%% 
%%   Returns: RetVal
%%
handle_call({OE_THIS, OE_Context, remove_all_constraints, []}, _, OE_State) ->
  corba:handle_call('CosNotifyFilter_Filter_impl', remove_all_constraints, [], OE_State, OE_Context, OE_THIS, false);

%%%% Operation: destroy
%% 
%%   Returns: RetVal
%%
handle_call({OE_THIS, OE_Context, destroy, []}, _, OE_State) ->
  corba:handle_call('CosNotifyFilter_Filter_impl', destroy, [], OE_State, OE_Context, OE_THIS, false);

%%%% Operation: match
%% 
%%   Returns: RetVal
%%   Raises:  CosNotifyFilter::UnsupportedFilterableData
%%
handle_call({OE_THIS, OE_Context, match, [Filterable_data]}, _, OE_State) ->
  corba:handle_call('CosNotifyFilter_Filter_impl', match, [Filterable_data], OE_State, OE_Context, OE_THIS, false);

%%%% Operation: match_structured
%% 
%%   Returns: RetVal
%%   Raises:  CosNotifyFilter::UnsupportedFilterableData
%%
handle_call({OE_THIS, OE_Context, match_structured, [Filterable_data]}, _, OE_State) ->
  corba:handle_call('CosNotifyFilter_Filter_impl', match_structured, [Filterable_data], OE_State, OE_Context, OE_THIS, false);

%%%% Operation: match_typed
%% 
%%   Returns: RetVal
%%   Raises:  CosNotifyFilter::UnsupportedFilterableData
%%
handle_call({OE_THIS, OE_Context, match_typed, [Filterable_data]}, _, OE_State) ->
  corba:handle_call('CosNotifyFilter_Filter_impl', match_typed, [Filterable_data], OE_State, OE_Context, OE_THIS, false);

%%%% Operation: attach_callback
%% 
%%   Returns: RetVal
%%
handle_call({OE_THIS, OE_Context, attach_callback, [Callback]}, _, OE_State) ->
  corba:handle_call('CosNotifyFilter_Filter_impl', attach_callback, [Callback], OE_State, OE_Context, OE_THIS, false);

%%%% Operation: detach_callback
%% 
%%   Returns: RetVal
%%   Raises:  CosNotifyFilter::CallbackNotFound
%%
handle_call({OE_THIS, OE_Context, detach_callback, [Callback]}, _, OE_State) ->
  corba:handle_call('CosNotifyFilter_Filter_impl', detach_callback, [Callback], OE_State, OE_Context, OE_THIS, false);

%%%% Operation: get_callbacks
%% 
%%   Returns: RetVal
%%
handle_call({OE_THIS, OE_Context, get_callbacks, []}, _, OE_State) ->
  corba:handle_call('CosNotifyFilter_Filter_impl', get_callbacks, [], OE_State, OE_Context, OE_THIS, false);



%%%% Standard gen_server call handle
%%
handle_call(stop, _, State) ->
    {stop, normal, ok, State};

handle_call(_, _, State) ->
    {reply, catch corba:raise(#'BAD_OPERATION'{minor=1163001857, completion_status='COMPLETED_NO'}), State}.


%%%% Standard gen_server cast handle
%%
handle_cast(stop, State) ->
    {stop, normal, State};

handle_cast(_, State) ->
    {noreply, State}.


%%%% Standard gen_server handles
%%
handle_info(Info, State) ->
    corba:handle_info('CosNotifyFilter_Filter_impl', Info, State).


code_change(OldVsn, State, Extra) ->
    corba:handle_code_change('CosNotifyFilter_Filter_impl', OldVsn, State, Extra).

